src/batches/models/BatchAdjustment.ts

```diff
@@ -4,7 +4,8 @@ import { InventoryStatus } from '../../inventories/enum/InventoryStatus';
 export interface BatchAdjustment {
   quantity: number;
   inventoryStatus: InventoryStatus;
-  idempotentKey: string;
+  idempotentKey?: string;
```

Comment:
I think we do one or the other here. I'm good with removing idempotent key here if it makes it easier to use.

Reply:
Do you mean dedup or idempotent?

Reply:
I think it would be better to keep the dedupe key because the other interface uses that as well. It's just a little confusing that you can provide both here.